api/main.go

```diff
@@ -175,6 +175,10 @@ type ConfigValue struct {
 	b   bool
 }
 
+func ConfigString(v string) ConfigValue   { return ConfigValue{typ: configTypeString, s: v} }
```

Comment:
This struct has moved to config.go, along with the Config struct and associated methods. Should it remain in api/main.go?

Reply:
Yeah, we talked about that. I didn't want to start moving stuff around though in order to avoid merge conflicts. I have rebased this code on top of the updated *plugin/complex_config_callback_v2* branch and moved these functions to `config.go`.

I'm beginning to wonder if it wouldn't be nicer if the config types were in their own package, `config`. `config.String()` would initialize a string `config.Value` (nee `api.ConfigValue`), for example. `api.Config` could become `config.Block`. It is unclear why `api.Unmarshaller` refers to configuration specifically – `config.Unmarshaller` doesn't have this problem. Etc. pp.

Reply:
I think `package config` would be ideal for the reasons you mention.